readme.md

```diff
@@ -71,12 +100,250 @@ array like some other http clients and it supports lazy streaming serialization
 
 ; supply a request map to customize other aspects of the request
 (http/get client "https://httbin.org/get" {:query-params {:a 1}})
+(http/get client "https://httpbin.org/anything" {:query-params {:a "foo" "b" ["1" "2"] :c 3}})
+
+;; custom headers (can mix keywords and strings)
+(http/get client "https://httpbin.org/anything" {:headers {"content-type" "text/plain" :accept :json}})
 
 ; make asynchronous calls by using callbacks
 (def request {})
 (def respond (fn [response] (println "response" response)))
 (def raise   (fn [exception] (.printStackTrace exception)))
 (http/get client "https://httbin.org/get" request respond raise)
+```
+
+### Creating a Client
+
+For a full list of client options see `clj-okhttp.core/create-client`. Explanation of the parameters
+can be found [here](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/).
+
+With that being said there are a few common options to look at:
+
+#### Connection Pool
+
+You may want to adjust the default connection pool to increase the number of connections or adjust the keep alive.
+
+```clojure
+{:connection-pool {:max-idle-connections 5 :keep-alive-duration 5 :time-unit java.util.concurrent.TimeUnit/MINUTES}}
+```
+
+#### Redirects
+
+By default the client will follow redirects and ssl redirects. This behavior can be adjusted if desired.
+
+```clojure
+{:follow-redirects true 
+ :follow-ssl-redirects true}
+```
+
+#### Timeouts
+
+By default the client will have a connect/read/write timeout of 10s and no call timeout.
+
+```clojure
+{:call-timeout 0 :connect-timeout 10000 :read-timeout 10000 :write-timeout 10000}
+```
+
+#### Allowing self-signed certificates
+
+While not recommended (useful for testing), you can ignore all self signed certificates via the `:insecure?` flag.
+
+```clojure
+{:insecure? true}
+```
+
+#### Cookies
+
+A custom `CookieJar` may be supplied. Below is an example of an in-memory CookieJar
+
+```clojure
+(def MemoryCookieJar
```

Comment:
I'd be open to just including a standard in-memory cookie jar impl so users don't have to recreate this if they want behavior "like a browser".